src/Summoner/Template.hs

```diff
@@ -0,0 +1,528 @@
+        {-# START_FILE test/Spec.hs #-}
+        main :: IO ()
+        main = putStrLn "Test suite not yet implemented"
+        $endLine
+        |]
+
+    createLib :: Text
+    createLib =
+        [text|
+        {-# START_FILE src/Lib.hs #-}
+        module Lib
+            ( someFunc
+            ) where
+
+        someFunc :: IO ()
+        someFunc = putStrLn "someFunc"
+        $endLine
+        |]
+
+    createOnlyExe :: Text
+    createOnlyExe =
+        [text|
+        {-# START_FILE app/Main.hs #-}
+        module Main where
+
+        main :: IO ()
+        main = putStrLn "Hello, world!"
+        $endLine
+        |]
+
+    createExe :: Text
+    createExe =
+        [text|
+        {-# START_FILE app/Main.hs #-}
+        module Main where
+
+        import Lib
+
+        main :: IO ()
+        main = someFunc
+        $endLine
+        |]
+
+    createBenchmark :: Text
+    createBenchmark =
+      [text|
+      {-# START_FILE benchmark/Main.hs #-}
+      import Criterion.Main
+
+      main :: IO ()
+      main = defaultMain [bench "const" (whnf const ())]
+      $endLine
+      |]
+
+    -- create README template
+    readme :: Text
+    readme =
+        [text|
+        {-# START_FILE README.md #-}
+        # $repo
+
+        [![Hackage]($hackageShield)]($hackageLink)
+        [![Build status](${travisShield})](${travisLink})
+        [![Windows build status](${appVeyorShield})](${appVeyorLink})
+        [![$license license](${licenseShield})](${licenseLink})
+
+        $description
+        $endLine
+        |]
+      where
+        hackageShield :: Text =
+          "https://img.shields.io/hackage/v/" <> repo <> ".svg"
+        hackageLink :: Text =
+          "https://hackage.haskell.org/package/" <> repo
+        travisShield :: Text =
+          "https://secure.travis-ci.org/" <> owner <> "/" <> repo <> ".svg"
+        travisLink :: Text =
+          "https://travis-ci.org/" <> owner <> "/" <> repo
+        appVeyorShield :: Text =
+          "https://ci.appveyor.com/api/projects/status/github/" <> owner <> "/" <> repo <> "?branch=master&svg=true"
+        appVeyorLink :: Text =
+          "https://ci.appveyor.com/project/" <> owner <> "/" <> repo
+        licenseShield :: Text =
+          "https://img.shields.io/badge/license-" <> T.replace "-" "--" license <> "-blue.svg"
+        licenseLink :: Text =
+          "https://github.com/" <> owner <> "/" <> repo <> "/blob/master/LICENSE"
+
+    -- create .gitignore template
+    gitignore :: Text
+    gitignore =
+        [text|
+        {-# START_FILE .gitignore #-}
+        ### Haskell
+        dist
+        dist-*
+        cabal-dev
+        *.o
+        *.hi
+        *.chi
+        *.chs.h
+        *.dyn_o
+        *.dyn_hi
+        *.prof
+        *.aux
+        *.hp
+        *.eventlog
+        .virtualenv
+        .hsenv
+        .hpc
+        .cabal-sandbox/
+        cabal.sandbox.config
+        cabal.config
+        cabal.project.local
+        .HTF/
```

Comment:
I think you also could add `.ghc.environment.*` to this list. Sometimes useful when you need to build project with `.cabal` as well. I needed it recently...